lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -36,94 +137,180 @@ class ConvertAtenMaxPool2dOp : public OpConversionPattern<AtenMaxPool2dOp> {
-      return rewriter.notifyMatchFailure(op, "only support kernel size ints");
-
-    Value falseValue = rewriter.create<arith::ConstantOp>(
-        loc, IntegerAttr::get(rewriter.getIntegerType(1), 0));
-    Value ceilModeFalse = rewriter.create<arith::CmpIOp>(
-        loc, arith::CmpIPredicate::eq, ceilMode, falseValue);
-    rewriter.create<cf::AssertOp>(
-        loc, ceilModeFalse,
-        rewriter.getStringAttr("only ceil_mode false is supported"));
+    // Contains the result of maxpool2d operation over the input.
+    Value maxPool2d;
+    if (failed(computeMaxPool2d<AtenMaxPool2dWithIndicesOp>(
+            op, rewriter, self, kernelSizeInts, strideInts, paddingInts,
+            dilationInts, maxPool2d)))
+      return rewriter.notifyMatchFailure(op, "unable to compute maxpool2d");
 
     SmallVector<int64_t, 4> paddingIncludingNC = {0, 0};
     paddingIncludingNC.insert(paddingIncludingNC.end(), paddingInts.begin(),
                               paddingInts.end());
     Value paddedInput = torch_to_linalg::getPaddedTensor(op, rewriter, self,
                                                          paddingIncludingNC);
 
-    Value N = getDimOp(rewriter, loc, self, 0);
-    Value C = getDimOp(rewriter, loc, self, 1);
-    Value H = getDimOp(rewriter, loc, self, 2);
-    Value W = getDimOp(rewriter, loc, self, 3);
-
-    SmallVector<Value> paddingIntValues =
-        getAsConstantIntValues(rewriter, loc, paddingInts);
-    SmallVector<Value> dilationIntValues =
-        getAsConstantIntValues(rewriter, loc, dilationInts);
-    SmallVector<Value> kernelSizeIntValues =
-        getAsConstantIntValues(rewriter, loc, kernelSizeInts);
-    SmallVector<Value> strideIntValues =
-        getAsConstantIntValues(rewriter, loc, strideInts);
-
-    Value Hout = torch_to_linalg::getOutputDimForConvOps(
-        rewriter, loc, H, paddingIntValues[0], dilationIntValues[0],
-        kernelSizeIntValues[0], strideIntValues[0]);
-    Value Wout = torch_to_linalg::getOutputDimForConvOps(
-        rewriter, loc, W, paddingIntValues[1], dilationIntValues[1],
-        kernelSizeIntValues[1], strideIntValues[1]);
-
-    // Initialize output tensor with smallest floating point value
-    Value outTensor = rewriter.create<linalg::InitTensorOp>(
-        loc, ValueRange{N, C, Hout, Wout}, elementType);
-    auto initialAttr = rewriter.getFloatAttr(
-        elementType,
-        APFloat::getSmallest(
-            elementType.cast<mlir::FloatType>().getFloatSemantics(),
-            /*Negative*/ true));
-    Value initValue = rewriter.create<arith::ConstantOp>(loc, initialAttr);
-    Value outTensorInitialized =
-        rewriter.create<linalg::FillOp>(loc, initValue, outTensor).getResult(0);
-
-    auto stridesAttr = rewriter.getI64VectorAttr(strideInts);
-    auto dilationAttr = rewriter.getI64VectorAttr(dilationInts);
+    SmallVector<Value> resultShape(getTensorSizes(rewriter, loc, maxPool2d));
+    Value indicesTensor = createZeroInitTensor(
```

Comment:
Need to init the tensor with -1 for detecting cases where max value appear multiple times in the window.